internal/backend/secrets/vault.go

```diff
@@ -0,0 +1,51 @@
+package secrets
+
+import (
+	"context"
+	"errors"
+
+	vault "github.com/hashicorp/vault/api"
+	"go.uber.org/zap"
+)
+
+type vaultSecrets struct {
+	client *vault.KVv2
+	logger *zap.Logger
+}
+
+// NewVaultSecrets initializes a client connection to HashiCorp Vault.
+func newVaultSecrets(l *zap.Logger, c *vaultConfig) (Secrets, error) {
```

Comment:
The rule of thumb is: `Accept interfaces, return types.`. This should be:

`func newVaultSecrets(l *zap.Logger, c *vaultConfig) (valutSecrets, error) {`

Reply:
that's a good rule of thumb, I fixed the internal objects I use (db/aws/vault stores) but not the external object other services requires

I think we have many places that returns interfaces instead of concrete types
but currently our fx configuration requires additional code if we'll return concrete types and not interfaces for the DI to work